[Feature]: 테스트 케이스로 factrial 함수 구현 중 문제

### 💡 Feature 설명
factrial 함수는 expression을 입력 받아서 len를 확인한 후 정수, 음수, 소수, 기호 등의 예외처리를 합니다. 근데 정수를 입력하는 테스트 케이스에서 AttributeError: 'int' object has no attribute 'replace' 에러가 뜨는데 왜 발생하는지 모르겠습니다. 아래는 테스트케이스입니다.

def factorial(expression):
    if len(str(expression).split()) > 1:
        return "[ERROR] Input Error"
    expression_no_space = expression.strip()
    if not str(expression_no_space).isdigit():
        return "[ERROR] Out Of Range"
    return "=" + str(math.factorial(int(expression_no_space)))

def testPositive(self):
        self.assertEqual(factorial(5),"=120")
def testZero(self):
        self.assertEqual(factorial(0),"=1")
def testNegative(self):
        self.assertEqual(factorial(-1),"[ERROR] Out Of Range")

### ⭐️ 우선순위

★★★★★

### 💁 예상 작업 기간

~12.15(금)

### 🙋실제 작업 기간(PR 작성 직전에 수정)

_No response_

--- a/main.py
+++ b/main.py
@@ -1,3 +1,6 @@
+import math
+import unittest
+
 def calculate(expression):
     try:
         return eval(expression)
@@ -28,18 +31,38 @@ def easterEgg(userInput):
     else:
         return "Not App"
 
+def factorial(expression):
+    if len(str(expression).split()) > 1:
+        return "[ERROR] Input Error"
+    expressionStriped = expression.strip()
+    if not str(expressionStriped).isdigit():
+        return "[ERROR] Out Of Range"
+    return "=" + str(math.factorial(int(expressionStriped)))
+
+class TestFactorialFunction(unittest.TestCase):
+    def testMultiple(self):
+        self.assertEqual(factorial("3 3"),"[ERROR] Input Error")
+        self.assertEqual(factorial("4 4 4"),"[ERROR] Input Error")
+    def testPositive(self):
+        self.assertEqual(factorial('5'),"=120")
+    def testZero(self):
+        self.assertEqual(factorial('0'),"=1")
+    def testNegative(self):
+        self.assertEqual(factorial('-1'),"[ERROR] Out Of Range")
+    def testNoneInteger(self):
+       self.assertEqual(factorial('+'),"[ERROR] Out Of Range")
+       self.assertEqual(factorial('a'),"[ERROR] Out Of Range")
+       self.assertEqual(factorial('0.1'),"[ERROR] Out Of Range")
 
 def main():
     expression = ""
     errorChecker = False
     compare = 0     #피연산자와 연산사 가려주는 변수
 
-
     while True:
         userInput = input().strip()
         easterEggValue = easterEgg(userInput)
         compare+=1
-
 
         if userInput == "=":
             # 주석해제하여 체크 / print(f"exp: {expression}\nerrChecker: {errorChecker}")
@@ -54,7 +77,11 @@ def main():
         elif easterEggValue != "Not App":
             print(f"[EVENT] \"{easterEggValue}\"")
             break
-    
+        if userInput == "!":
+            print(factorial(expression))
+            
+            break
+
         #elif isFloat(userInput) or isDivision(userInput):
         #   errorChecker = True
         else:
@@ -65,8 +92,8 @@ def main():
                 if isDivision(userInput):
                     errorChecker = True
             # 주석해제하여 체크 / print(f"before add: {expression}")
-            expression += userInput
+            expression += userInput + " "  
             # 주석해제하여 체크 / print(f"after add: {expression}")
 if __name__ == "__main__":
-    main()
-
+    unittest.main()
+    #main()

--- a/main.py
+++ b/main.py
@@ -1,5 +1,5 @@
-import math
 import unittest
+import math
 
 def calculate(expression):
     try:
@@ -7,12 +7,14 @@ def calculate(expression):
     except Exception as e:
         return f"Error: {e}"
 
+
 def isDivision(userInput):
     if userInput in ("+", "-", "*"):
         return False
-    elif userInput == '/':      #10번 줄 없애도 됨
-        return True             #11번 줄 없애도 됨
+    elif userInput == '/':  # 10번 줄 없애도 됨
+        return True  # 11번 줄 없애도 됨
     return True
+
 
 def isFloat(userInput):
     if "." in userInput:
@@ -23,6 +25,7 @@ def isFloat(userInput):
             return False
     return False
 
+
 def easterEgg(userInput):
     if userInput == "5252":
         return "정종욱 교수님을 총장으로"
@@ -30,6 +33,7 @@ def easterEgg(userInput):
         return "전북대 개교기념일입니다."
     else:
         return "Not App"
+
 
 def factorial(expression):
     if len(str(expression).split()) > 1:
@@ -39,30 +43,36 @@ def factorial(expression):
         return "[ERROR] Out Of Range"
     return "=" + str(math.factorial(int(expressionStriped)))
 
+
 class TestFactorialFunction(unittest.TestCase):
     def testMultiple(self):
-        self.assertEqual(factorial("3 3"),"[ERROR] Input Error")
-        self.assertEqual(factorial("4 4 4"),"[ERROR] Input Error")
+        self.assertEqual(factorial("3 3"), "[ERROR] Input Error")
+        self.assertEqual(factorial("4 4 4"), "[ERROR] Input Error")
+
     def testPositive(self):
-        self.assertEqual(factorial('5'),"=120")
+        self.assertEqual(factorial("5"), "=120")
+
     def testZero(self):
-        self.assertEqual(factorial('0'),"=1")
+        self.assertEqual(factorial("0"), "=1")
+
     def testNegative(self):
-        self.assertEqual(factorial('-1'),"[ERROR] Out Of Range")
+        self.assertEqual(factorial("-1"), "[ERROR] Out Of Range")
+
     def testNoneInteger(self):
-       self.assertEqual(factorial('+'),"[ERROR] Out Of Range")
-       self.assertEqual(factorial('a'),"[ERROR] Out Of Range")
-       self.assertEqual(factorial('0.1'),"[ERROR] Out Of Range")
+        self.assertEqual(factorial('+'), "[ERROR] Out Of Range")
+        self.assertEqual(factorial('a'), "[ERROR] Out Of Range")
+        self.assertEqual(factorial('0.1'), "[ERROR] Out Of Range")
+
 
 def main():
     expression = ""
     errorChecker = False
-    compare = 0     #피연산자와 연산사 가려주는 변수
+    compare = 0  # 피연산자와 연산사 가려주는 변수
 
     while True:
         userInput = input().strip()
         easterEggValue = easterEgg(userInput)
-        compare+=1
+        compare += 1
 
         if userInput == "=":
             # 주석해제하여 체크 / print(f"exp: {expression}\nerrChecker: {errorChecker}")
@@ -79,21 +89,22 @@ def main():
             break
         if userInput == "!":
             print(factorial(expression))
-            
             break
 
-        #elif isFloat(userInput) or isDivision(userInput):
+        # elif isFloat(userInput) or isDivision(userInput):
         #   errorChecker = True
         else:
-            if compare % 2: #정수 입력
+            if compare % 2:  # 정수 입력
                 if isFloat(userInput):
                     errorChecker = True
-            elif not (compare % 2): #연산자 입력
+            elif not (compare % 2):  # 연산자 입력
                 if isDivision(userInput):
                     errorChecker = True
             # 주석해제하여 체크 / print(f"before add: {expression}")
             expression += userInput + " "  
             # 주석해제하여 체크 / print(f"after add: {expression}")
+
+
 if __name__ == "__main__":
     unittest.main()
     #main()

